src/Parser/AfflictionWarlock/Modules/Features/CooldownTracker.js

```diff
@@ -11,44 +13,91 @@ class CooldownTracker extends CoreCooldownTracker {
         BUILT_IN_SUMMARY_TYPES.DAMAGE,
       ],
     },
-    //for the sake of completeness, typically unused
+  ];
+
+  //for the sake of completeness, typically none of these are used
+  static castCooldowns = [
     {
       spell: SPELLS.SUMMON_INFERNAL_UNTALENTED,
+      duration: 25,
       summary: [
         BUILT_IN_SUMMARY_TYPES.DAMAGE,
       ],
     },
     {
       spell: SPELLS.SUMMON_DOOMGUARD_UNTALENTED,
+      duration: 25,
       summary: [
         BUILT_IN_SUMMARY_TYPES.DAMAGE,
       ],
     },
     {
       spell: SPELLS.GRIMOIRE_IMP,
+      duration: 25,
       summary: [
         BUILT_IN_SUMMARY_TYPES.DAMAGE,
       ],
     },
     {
       spell: SPELLS.GRIMOIRE_VOIDWALKER,
+      duration: 25,
       summary: [
         BUILT_IN_SUMMARY_TYPES.DAMAGE,
       ],
     },
     {
       spell: SPELLS.GRIMOIRE_SUCCUBUS,
+      duration: 25,
       summary: [
         BUILT_IN_SUMMARY_TYPES.DAMAGE,
       ],
     },
     {
       spell: SPELLS.GRIMOIRE_FELHUNTER,
+      duration: 25,
       summary: [
         BUILT_IN_SUMMARY_TYPES.DAMAGE,
       ],
     },
   ];
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    const cooldownSpell = this.constructor.castCooldowns.find(cooldownSpell => cooldownSpell.spell.id === spellId);
+    if (cooldownSpell) {
+      // adding the fixed cooldown, now we need to remove it from activeCooldowns too
+      const cooldown = this.addFixedCooldown(cooldownSpell, event.timestamp);
+      this.activeCooldowns.push(cooldown);
+      debug && console.log(`%cCooldown started: ${cooldownSpell.spell.name}`, 'color: green', cooldown);
+    }
+    // super.on_byPlayer_cast(event) would call trackEvent anyway
+    this.trackEvent(event);
+  }
+
+  addFixedCooldown(cooldownSpell, timestamp) {
```

Comment:
This should be private; prefixed with an underscore (addCooldown too, but that's old code).